### PR TITLE
Profile tag improvements and target user authentication

### PR DESCRIPTION
**Pull request by lalithr95**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [x] reviewed/confirmed/tested by another contributor or maintainer
* [x] `development.sqlite.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by lalithr95** [comment]
This PR will add few validations for tag value as unit tests which were missed in #576 . Target user authentication is another feature where admin can remove or add tags to other users where the uid should be target user. Tests are written for the same.

cc @jywarren Ready for merge.

File: test/functional/user_tags_controller_test.rb [hunk 1]
```diff
@@ -41,22 +41,41 @@ def setup
...
+    post :create, :id => rusers(:bob).username, :type => "role", :value => "Organizer"
+    user_tag = UserTag.where(uid: rusers(:bob).id).last
+    user_tags_count = UserTag.where(uid: rusers(:bob).id).count
+    # Delete above tag
+    UserSession.create(rusers(:jeff))
+    post :delete, :id => user_tag.id
+    assert_equal user_tags_count -1, UserTag.where(uid: rusers(:bob).id).count
+  end
+
+  test "admin should create new tags for normal user" do
```
Comment (jywarren) [inline test/functional/user_tags_controller_test.rb]: Can you counter-test this, by confirming that *non-admins* can NOT create tags for others, but rename the ones above to show that this IS possible for your own tags? Thanks!!

**Comment by jywarren** [comment]
Awesome, a few additional requests for tests here, and I'm not sure where in your controller you are preventing non-admins from making/changing tags on other users?

**Comment by lalithr95** [comment]
@jywarren Thanks for the note. Seriously don't know how I missed such big flaw. Updated at the backend controller and necessary tests. Let me know your views.

**Comment by jywarren** [comment]
Hmm, i get one unit test failure:

````
========================================================================================================================================================================
Failure:
  <["Value contains invalid input"]> expected but was
  <[]>.
test_should_not_contains_special_characters_in_value(UserTagTest)
test/unit/user_tag_test.rb:51:in `block (2 levels) in <class:UserTagTest>'
test/unit/user_tag_test.rb:45:in `each'
test/unit/user_tag_test.rb:45:in `block in <class:UserTagTest>'
========================================================================================================================================================================
````

Can you check that out? Thanks!

**Comment by lalithr95** [comment]
Updated unit test. All tests are passing. Ready for merge Thanks !

**Comment by jywarren** [comment]
Rebasing and running tests

**Comment by jywarren** [comment]
Manually merged -- thanks!

**Comment by lalithr95** [comment]
@jywarren Thanks !

**Pull request closed without merging**